private/bufpkg/bufconfig/buf_policy_yaml_file_test.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020-2025 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bufconfig
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestReadWriteBufPolicyYAMLFileRoundTrip(t *testing.T) {
+	t.Parallel()
+
+	testReadWriteBufPolicyYAMLFileRoundTrip(
+		t,
+		// input
+		`version: v2
+`,
+		// expected output
+		`version: v2
+`,
+	)
+
+	testReadWriteBufPolicyYAMLFileRoundTrip(
+		t,
+		// input
+		`version: v2
```

Comment:
Curious, this is the first policy file version, why we're starting at v2?

Reply:
The file will consume v2 lint/breaking config options, so we went with v2 to clarify its related to the v2 spec.